docs/source/test_network.md

```diff
@@ -0,0 +1,585 @@
+be used only as a tool for education and testing. It should not be used as a
+template for deploying a production network.
+
+The sample network deploys a Fabric network with Docker Compose. Because the
+nodes are isolated within a Docker Compose network, the test network is not
+configured to connect to other running fabric nodes.
+
+**Note:** These instructions have been verified to work against the
+latest stable Docker images and the pre-compiled setup utilities within the
+supplied tar file. If you run these commands with images or tools from the
+current master branch, it is possible that you will encounter errors.
+
+## Before you begin
+
+Before you can run the test network, you need to clone the `fabric-samples`
+repository and download the Fabric images. Make sure that that you have installed
+the [Prerequisites](prereqs.html) and [Installed the Samples, Binaries and Docker Images](install.html).
+
+## Bring up the test network
+
+You can find the scripts to bring up the network in the `test-network` directory
+of the ``fabric-samples`` repository. Navigate to the test network directory by
+using the following command:
+```
+cd fabric-samples/test-network
+```
+
+In this directory, you can find an annotated script, ``network.sh``, that stands
+up a Fabric network using the Docker images on your local machine. You can run
+``network.sh -h`` to print the script help text:
+
+```
+Usage:
+  network.sh <Mode> [Flags]
+    <Mode>
+      - 'up' - Bring up fabric orderer and peer nodes. No channel is created
+      - 'up createChannel' - bring up fabric network with one channel
+      - 'createChannel' - create and join a channel after the network is created
+      - 'deployCC' - Deploy a chainocode on the channel
+      - 'down' - clear the network with docker-compose down
+      - 'restart' - restart the network
+
+    Flags:
+    -ca <use CAs> -  create Certificate Authorities to generate the crypto material
+    -c <channel name> - channel name to use (defaults to "mychannel")
+    -s <dbtype> - the database backend to use: goleveldb (default) or couchdb
+    -r <max retry> - CLI times out after certain number of attempts (defaults to 5)
+    -d <delay> - delay duration in seconds (defaults to 3)
+    -l <language> - the programming language of the chaincode to deploy: go (default), javascript, or java
+    -v <version>  - chaincode version. Must be a round number, 1, 2, 3, etc
+    -i <imagetag> - the tag to be used to launch the network (defaults to "latest")
+    -verbose - verbose mode
+  network.sh -h (print this message)
+
+ Possible Mode and flags
+  network.sh up -ca -c -r -d -s -i -verbose
+  network.sh up createChannel -ca -c -r -d -s -i -verbose
+  network.sh createChannel -c -r -d -verbose
+  network.sh deployCC -l -v -r -d -verbose
+
+ Taking all defaults:
+	network.sh up
+
+ Examples:
+  network.sh up createChannel -ca -c mychannel -s couchdb -i 1.4.0
```

Comment:
Channel creation is failing while executing this command. Could be timeout issue.. Add re-try or sleep after spinup the fabric network.

Reply:
This happens because the ordering service has not started. Added retry to the create command to solve.